packages/components/src/FormatFile/FormatFile.tsx

```diff
@@ -1,29 +1,72 @@
 import React, { useState } from "react";
 import filesize from "filesize";
+import classNames from "classnames";
 import { IconNames } from "@jobber/design";
 import styles from "./FormatFile.css";
-import { Button } from "../Button";
-import { Icon } from "../Icon";
+import { FormatFileDeleteButton } from "./FormatFileDeleteButton";
+import { DisplaySize, sizeToDimensions } from "./sizeToDimensions";
+import { ImageWithoutSource } from "./ImageWithoutSource";
 import { Typography } from "../Typography";
 import { ProgressBar } from "../ProgressBar";
 import { FileUpload } from "../InputFile";
 
+const progressBar = (file: FileUpload) => (
+  <div className={styles.progress}>
+    <ProgressBar
+      size="small"
+      currentStep={file.progress * 100}
+      totalSteps={100}
+    />
+  </div>
+);
+
 interface FormatFileProps {
   /**
    * File upload details object. (See FileUpload type.)
    */
   readonly file: FileUpload;
 
+  /**
+   * To display as either a file row or thumbnail
+   *
+   * @default "expanded"
+   */
+  readonly display?: "expanded" | "compact";
+
+  /**
+   * The base dimensions of the thumbnail
+   *
+   * @default "default"
+   */
+  readonly displaySize?: DisplaySize;
+
+  /**
+   * Function to execute when format file is clicked
+   */
+  onClick?(event: React.MouseEvent<HTMLDivElement>): void;
+
   /**
    * onDelete callback - this function will be called when the delete action is triggered
    */
   onDelete?(): void;
 }
 
-export function FormatFile({ file, onDelete }: FormatFileProps) {
+export function FormatFile({
+  file,
+  display = "expanded",
+  displaySize = "default",
+  onDelete,
+  onClick,
+}: FormatFileProps) {
   const [imageSource, setImageSource] = useState<string>();
   const isComplete = file.progress >= 1;
 
+  const thumbnailDimensions = sizeToDimensions[displaySize];
+  const [deleteConfirmationOpen, setDeleteConfirmationOpen] = useState(false);
+  const [showDeleteButton, setShowDeleteButton] = useState(
+    display === "compact" && displaySize === "default" ? false : true,
+  );
+
   const iconName = getIconNameFromType(file.type);
   const fileSize = getHumanReadableFileSize(file.size);
```

Comment:
I think this should be named something to describe what it does better. style is too generic in my opinion